WordPress/Classes/ViewRelated/Post/PostEditorState.swift

```diff
@@ -0,0 +1,292 @@
+import Foundation
+
+// Taken from AbstractPost.m - this should get a better home than here
+enum PostStatus: String {
+    case draft = "draft"
+    case pending = "pending"
+    case publishPrivate = "private"
+    case publish = "publish"
+    case scheduled = "future"
+    case trash = "trash"
+    case deleted = "deleted" // Returned by wpcom REST API when a post is permanently deleted.
+}
+
+public enum PostEditorAction {
+    case save
+    case schedule
+    case publish
+    case update
+    case submitForReview
+
+    var publishActionLabel: String {
+        switch self {
+        case .publish:
+            return NSLocalizedString("Publish", comment: "Publish button label.")
+        case .save:
+            return NSLocalizedString("Save", comment: "Save button label (saving content, ex: Post, Page, Comment).")
+        case .schedule:
+            return NSLocalizedString("Schedule", comment: "Schedule button, this is what the Publish button changes to in the Post Editor if the post has been scheduled for posting later.")
+        case .submitForReview:
+            return NSLocalizedString("Submit for Review", comment: "Submit for review button label (saving content, ex: Post, Page, Comment).")
+        case .update:
+            return NSLocalizedString("Update", comment: "Update button label (saving content, ex: Post, Page, Comment).")
+        }
+    }
+
+    var publishingActionLabel: String {
+        switch self {
+        case .publish:
+            return NSLocalizedString("Publishing...", comment: "Text displayed in HUD while a post is being published.")
+        case .save:
+            return NSLocalizedString("Saving...", comment: "Text displayed in HUD while a post is being saved as a draft.")
+        case .schedule:
+            return NSLocalizedString("Scheduling...", comment: "Text displayed in HUD while a post is being scheduled to be published.")
+        case .submitForReview:
+            return NSLocalizedString("Submitting for Review...", comment: "Text displayed in HUD while a post is being submitted for review.")
+        case .update:
+            return NSLocalizedString("Updating...", comment: "Text displayed in HUD while a draft or scheduled post is being updated.")
+        }
+    }
+
+    var isPostPostShown: Bool {
```

Comment:
Looks like this isn't used yet and I might be misunderstanding its use, but with the hybrid editor we also show post-post for scheduled posts.